pkg/controller/controller.go

```diff
@@ -289,3 +289,15 @@ func VMIActivePodsCount(vmi *v1.VirtualMachineInstance, vmiPodInformer cache.Sha
 
 	return running
 }
+
+func GeneratePatchBytes(ops []string) []byte {
+	opsStr := "["
+	for idx, entry := range ops {
+		sep := ", "
+		if len(ops)-1 == idx {
+			sep = "]"
+		}
+		opsStr = fmt.Sprintf("%s%s%s", opsStr, entry, sep)
```

Comment:
This might be suboptimal. we're copying parts of the string over and over again as the loop iterates.

Would it be less copying if we created a new slice of []string that had the separators included then just call string.Join()?

This might not be a huge deal though. These arrays are usually just a handful of operations long.

Reply:
I agree, wouldn't a simple `"[" + strings.Join(ops, ",") + "]"` have the same result?

Reply:
yup, fixed